### PR TITLE
[TRRFFDA-644] Make primary carer email unique

### PR DESCRIPTION
**Pull request by alintreznai**

### CODE REVIEW

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+def primary_carer_unique_email(apps, schema_editor):
+    PrimaryCarer = apps.get_model('mnd', 'PrimaryCarer')
+    PrimaryCarerRelationship = apps.get_model('mnd', 'PrimaryCarerRelationShip')
+    Patient = apps.get_model('patients', 'Patient')
+    CustomUser = apps.get_model('groups', 'CustomUser')
+    carer_email_dict = defaultdict(list)
+    for pc in PrimaryCarer.objects.all().order_by('id'):
+        carer_email_dict[pc.email.lower()].append(pc)
+    multiple_carers = { email: carers for email, carers in carer_email_dict.items() if len(carers) > 1}
+    print(f"Found {len(multiple_carers)} email addresses with multiple carers")
```
Comment (ScriptSmith) [inline mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py]: Are these print statements for debugging purposes, or intended to be in the final migration? I would think it's a bad idea to include them.

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+from collections import defaultdict
+import json
+from django.db import migrations, models
+
+
+
+def primary_carer_unique_email(apps, schema_editor):
+    PrimaryCarer = apps.get_model('mnd', 'PrimaryCarer')
+    PrimaryCarerRelationship = apps.get_model('mnd', 'PrimaryCarerRelationShip')
```
Comment (ScriptSmith) [inline mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py]: Should be `PrimaryCarerRelationship`?

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from collections import defaultdict
+import json
+from django.db import migrations, models
+
+
+
+def primary_carer_unique_email(apps, schema_editor):
```
Comment (ScriptSmith) [inline mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py:8]: Do we need this migration? What does django normally do when you make an existing field unique?

If the field wasn't already functionally unique, then I imagine there would be a runtime error from postgres. But can't we simplify this migration by manually removing non-unique carers (if there are any)?

If there were carers in production then I'd say we should do it properly, and I'm happy to do it this way if it's a good idea.

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py (hunk 1 above)
Comment (alintreznai) [reply mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py]: Removed print statements.

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py (hunk 2 above)
Comment (alintreznai) [reply mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py]: Good catch !. Removed

File: mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py (hunk 3 above)
Comment (alintreznai) [reply mnd/mnd/migrations/0017_primary_carer_email_unique_data_migration.py:8]: This was implemented assuming there are non-unique emails in production. 
The field wasn't already functionally unique as you could theoretically add two carers with the same email address ("email1@test.com" and "Email1@test.com" were treated as different addresses).
I still think It's a good idea to use a migration as it's more explicit and it's less prone to error than having to manually fix entries in the db.

**Pull request merged**